test/AWS.Deploy.Orchestration.UnitTests/DeployedApplicationQueryerTests.cs

```diff
@@ -0,0 +1,170 @@
+            Assert.Empty(result);
+        }
+
+        [Fact]
+        public async Task GetExistingDeployedApplications_NoStackDescriptionPrefix()
+        {
+            var stack = new Stack
+            {
+                Tags = new List<Tag>() { new Tag {
+                    Key = CloudFormationIdentifierConstants.STACK_TAG,
+                    Value = "AspNetAppEcsFargate"
+                } },
+                StackStatus = StackStatus.CREATE_COMPLETE,
+                StackName = "Stack1"
+            };
+
+            _mockAWSResourceQueryer
+                .Setup(x => x.GetCloudFormationStacks())
+                .Returns(Task.FromResult(new List<Stack>() { stack }));
+
+            var deployedApplicationQueryer = new DeployedApplicationQueryer(_mockAWSResourceQueryer.Object);
+
+            var result = await deployedApplicationQueryer.GetExistingDeployedApplications(new List<Recommendation>());
+            Assert.Empty(result);
+        }
+
+        [Fact]
+        public async Task GetExistingDeployedApplications_DeleteStatus()
+        {
+            var stack = new Stack
+            {
+                Tags = new List<Tag>() { new Tag {
+                    Key = CloudFormationIdentifierConstants.STACK_TAG,
+                    Value = "AspNetAppEcsFargate"
+                } },
+                Description = CloudFormationIdentifierConstants.STACK_DESCRIPTION_PREFIX,
+                StackStatus = StackStatus.DELETE_IN_PROGRESS,
+                StackName = "Stack1"
+            };
+
+            _mockAWSResourceQueryer
+                .Setup(x => x.GetCloudFormationStacks())
+                .Returns(Task.FromResult(new List<Stack>() { stack }));
+
+            var deployedApplicationQueryer = new DeployedApplicationQueryer(_mockAWSResourceQueryer.Object);
+
+            var result = await deployedApplicationQueryer.GetExistingDeployedApplications(new List<Recommendation>());
+            Assert.Empty(result);
+        }
+
+        [Fact]
+        public async Task GetExistingDeployedApplications_RollbackCompleteStatus()
+        {
+            var stack = new Stack
+            {
+                Tags = new List<Tag>() { new Tag {
+                    Key = CloudFormationIdentifierConstants.STACK_TAG,
+                    Value = "AspNetAppEcsFargate"
+                } },
+                Description = CloudFormationIdentifierConstants.STACK_DESCRIPTION_PREFIX,
+                StackStatus = StackStatus.ROLLBACK_COMPLETE,
+                StackName = "Stack1"
+            };
+
+            _mockAWSResourceQueryer
+                .Setup(x => x.GetCloudFormationStacks())
+                .Returns(Task.FromResult(new List<Stack>() { stack }));
+
+            var deployedApplicationQueryer = new DeployedApplicationQueryer(_mockAWSResourceQueryer.Object);
+
+            var result = await deployedApplicationQueryer.GetExistingDeployedApplications(new List<Recommendation>());
+            Assert.Empty(result);
+        }
```

Comment:
We are asserting the same thing again, it is good candidate of data driven test. i.e. Theory test in xUnit.

https://xunit.net/faq/theory-data-stability-in-vs

Reply:
I updated it.